src/docfx/build/Build.cs

```diff
@@ -15,48 +15,53 @@ internal static class Build
     {
         public static async Task Run(string docsetPath, CommandLineOptions options, Report report)
         {
-            var errors = new List<Error>();
+            using (var gitCommitProvider = new GitCommitProvider())
```

Comment:
This provider's lifecycle is expanded from a pre-step to nearly whole build, so that it cannot be exposed earlier. Will this impact performance, especially in the first run when commit cache is not present?

Reply:
This is exactly the structure we are looking for 💯 💯 💯 👍 👍 👍